internal/blocksync/reactor.go

```diff
@@ -560,6 +560,8 @@ func (r *Reactor) poolRoutine(ctx context.Context, stateSynced bool, blockSyncCh
 			if err == nil && state.ConsensusParams.ABCI.VoteExtensionsEnabled(extCommit.Height) {
 				// if vote extensions were required at this height, ensure they exist.
 				err = extCommit.EnsureExtensions()
+			} else if err == nil && !state.ConsensusParams.ABCI.VoteExtensionsEnabled(extCommit.Height) {
+				extCommit.StripExtensions()
```

Comment:
Careful here, `StripExtensions` is read-only. It returns a `Commit` structure.
I realize the name `StripExtensions` is somewhat misleading... probably something like `ToCommit` would be clearer

Reply:
Actually, I see you renamed the function precisely to `ToCommit`, and `StripExtensions` _does indeed_ strip the extensions, so please disregard the comment above

Reply:
Yeah, since we actually need to remove them from this type I figured the rename made sense.